### DISCUSSION TITLE
Azure DevOPS as an external datasource

### DISCUSSION THREAD

**Discussion by ziggekatten** [netbox-community/netbox#12494] [Help Wanted!] [answered]
The 3.5.0 feature to use Git as an source for custom scripts and reports are something I have been looking forward to, but I struggle to get this to work. Now, we do not use GitHub, but instead we have an on prem Azure DevOps (former TFS) using the Git standard protocol. Works just as GitHub using the git client and so forth.

Now, the first thing is the fact that Netbox stores the password for the user i clear text, and as we use Active Directory and personal accounts to access Azure DevOps UI, we can't use username/password. Instead, as we normally do, we create an user token with read permissions, and use that token as password when authenticating with a git client when needing to accessing code. This works fine from an standard git client in for example Linux.

So, I set up an source, using username and the token as password, master branch. But going into Reports, and selecting the source, I get no files in the drop down. 

My problem now is that I can't find anything in any logs where Netbox tries to actually access our repo. So I have really an hard time troubleshoot this. Where can I see this, or can I trigger an access somehow?

Lastly, password field should really now be shown in clear text, and as it also shows up in the change log, there is an potential security risk here. Should maybe end up in an Feature Request to change this behaviour

**Comment by ziggekatten** [comment, marked as answer]
Well, one should also do an: manage.py syncdatasource --all

Works like an charm after sync is done to add the source for the reports.

**Reply by v0tti** [reply]
If you do not see a `Sync` button for the data source you might be missing the `sync` action in your NetBox permissions.
